### PR TITLE
Ispn 3549 Migrate cloud cache store to the 7.0 cachestore API

### PR DESCRIPTION
**Pull request by vjuranek**
https://issues.jboss.org/browse/ISPN-3549

### CODE REVIEW

**Comment by snotling** [comment]
I'm glad to see this feature taking shape!

File: src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser [hunk 1]
```diff
@@ -1,3 +1 @@
-org.infinispan.loaders.cloud.configuration.CloudCacheStoreConfigurationParser52
-org.infinispan.loaders.cloud.configuration.CloudCacheStoreConfigurationParser53
-org.infinispan.loaders.cloud.configuration.CloudCacheStoreConfigurationParser60
+org.infinispan.persistence.cloud.configuration.CloudStoreConfigurationParser60
```
Comment (galderz) [inline src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser]: Shouldn't be 70 or 71 ending?

File: src/main/java/org/infinispan/persistence/cloud/configuration/Attribute.java [hunk 2]
```diff
@@ -7,27 +7,35 @@
  * Enumerates the attributes used by the Cloud cache store configuration
  *
  * @author Tristan Tarrant
- * @since 5.2
+ * @author Damiano Albani
+ * @since 6.0
  */
 public enum Attribute {
```
Comment (galderz) [inline src/main/java/org/infinispan/persistence/cloud/configuration/Attribute.java]: You've removed quite a few attributes here, why is that exactly?

File: src/main/java/org/infinispan/persistence/cloud/configuration/Attribute.java [hunk 3]
```diff
@@ -7,27 +7,35 @@
...
-   MAX_CONNECTIONS("maxConnections"),
-   PASSWORD("password"),
-   PROXY_HOST("proxyHost"),
-   PROXY_PORT("proxyPort"),
-   REQUEST_TIMEOUT("requestTimeout"),
-   SECURE("secure"),
-
+   CREDENTIAL("credential"),
+   CONTAINER("container"),
+   KEY_TO_STRING_MAPPER("key2StringMapper"),
```
Comment (galderz) [inline src/main/java/org/infinispan/persistence/cloud/configuration/Attribute.java]: Can you please follow same convention used as for JDBC cache store? It's

    KEY_TO_STRING_MAPPER("key-to-string-mapper")

File: src/main/resources/schema/infinispan-cachestore-cloud-config-6.0.xsd [hunk 4]
```diff
@@ -10,89 +11,39 @@
...
+                TODO
               </xs:documentation>
             </xs:annotation>
           </xs:attribute>
-          <xs:attribute name="maxConnections" type="xs:int" default="10">
+          <xs:attribute name="container" type="xs:string">
             <xs:annotation>
               <xs:documentation>
-                The maximum number of concurrent connections to make to the cloud provider. Defaults to 10.
+                TODO
```
Comment (galderz) [inline src/main/resources/schema/infinispan-cachestore-cloud-config-6.0.xsd]: Can you sort out these TODOs please? Thx :)

**Comment by galderz** [comment]
@vjuranek Have you done any testing with existing cloud cache stores, e.g. Amazon...etc? IOW, whether you've done testing beyond the in-memory version used in unit testing. I'm asking because if not done yet, it would show whether there are any limitations with the xml attribute reduction.

File: src/main/java/org/infinispan/persistence/cloud/configuration/CloudStoreConfigurationParser60.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+*
+* CloudStoreConfigurationParser60.
+*
+* @author Galder Zamarreño
+* @author Damiano Albani
+* @since 6.0
+*/
+public class CloudStoreConfigurationParser60 implements ConfigurationParser {
```
Comment (tristantarrant) [inline src/main/java/org/infinispan/persistence/cloud/configuration/CloudStoreConfigurationParser60.java]: We don't need a 6.0 parser

File: src/main/java/org/infinispan/persistence/cloud/configuration/CloudStoreConfigurationParser70.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+* CloudStoreConfigurationParser70.
+* 
+* @since 7.0
+*/
+@MetaInfServices
+@Namespaces({
+      @Namespace(uri = "urn:infinispan:config:store:cloud:7.0",
+                 root = CloudStoreConfigurationParser70.ROOT_ELEMENT),
+})
+public class CloudStoreConfigurationParser70 implements ConfigurationParser {
```
Comment (tristantarrant) [inline src/main/java/org/infinispan/persistence/cloud/configuration/CloudStoreConfigurationParser70.java]: We don't need a 7.0 parser, make it 7.2 (as that's going to be our next release)

File: src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser (hunk 1 above)
Comment (tristantarrant) [inline src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser]: This file should be removed and generated by the @MetaInfServices annotation

**Comment by vjuranek** [comment]
Thanks for the review, all comments should be addressed now.

@galderz I've tested so far only with `transient` and `file` providers. I''m going to test it with at least OpenStack in near future. 

Removed attributes are not mostly applicable (and AFAICT most of them were not actually used even with previous version of the cache store). The only attribute which IMHO makes sense was compression of stored payload. I can add include it if you like, but I would prefer to add it later on, once I test it with some real cloud providers (as there could be some strange issues introduced by compression, which could eventually make debugging hard)

**Comment by galderz** [comment]
Please do not remove compression. This was specifically added as a result of a customer request (see [ISPN-357](https://issues.jboss.org/browse/ISPN-357)). The main reason is was added is $$$. If uncompressed, you can incur in added data storage costs. Compression brings those down.

**Comment by galderz** [comment]
Apart from adding back compression, do you want to test other providers before pulling this PR? Or once compression is in, do you want me to integrate and you'll do further testing later?

**Comment by vjuranek** [comment]
@galderz well, payload compression is restored (with some minor changes - I switched from bzip to gzip to avoid additional dependencies and also add check of MD5 of the payload).

As for merging, if you think there's any chance that someone from the community will test it when it gets merge, plase integrate, otherwise it can wait until I test it myself (and eventually fix discovered issues)

**Comment by galderz** [comment]
Thanks for adding compression. Yeah, better use GZIP which adds no dependencies.

Let's merge it as is since there's already quite a bit in this.

**Comment by galderz** [comment]
Integrated, thanks @vjuranek !!

**Pull request closed without merging**